unix/modsocket.c

```diff
@@ -99,6 +99,11 @@ STATIC mp_uint_t socket_read(mp_obj_t o_in, void *buf, mp_uint_t size, int *errc
     mp_int_t r = read(o->fd, buf, size);
     if (r == -1) {
         *errcode = errno;
+
+        if (*errcode == EAGAIN) {
```

Comment:
This errno conversion is not correct for a case of non-blocking socket (for it, EAGAIN should remain EAGAIN).